app/javascript/channels/api/v1/race_channel.js

```diff
@@ -0,0 +1,263 @@
+const moment = require('moment')
+
+import consumer from '../../consumer'
+import { createAlert, showButton, hideButton } from '../../../dom_helpers'
+import { ts } from '../../../time'
+
+let raceSubscription
+let isConnected = false
+
+document.addEventListener('turbolinks:load', () => {
+  if (!window.gon || !window.gon.race) {
+    return
+  }
+
+  raceSubscription = consumer.subscriptions.create({
+    channel:    'Api::V1::RaceChannel',
+    race_id:    window.gon.race.id,
+    race_type:  window.gon.race.type,
+    join_token: window.gon.race.join_token
+  }, {
+    connected() {
+      isConnected = true
+      const alert = document.getElementById('disconnected-alert')
+      if (alert) {
+        alert.parentNode.removeChild(alert)
+      }
+
+      const forfeitTime = localStorage.getItem(`race-${window.gon.race.id}-forfeit-time`)
+      if (forfeitTime) {
+        this.forfeit(parseFloat(forfeitTime))
+        localStorage.removeItem(`race-${window.gon.race.id}-forfeit-time`)
+      }
+
+      const doneTime = localStorage.getItem(`race-${window.gon.race.id}-done-time`)
+      if (doneTime) {
+        this.done(parseFloat(doneTime))
+        localStorage.removeItem(`race-${window.gon.race.id}-done-time`)
+      }
+    },
```

Comment:
This is a super cool idea!  It should matter if the event exists though, since we could just emit our own even upon connection in our channel right?  Then we just listen for our own custom event and handle it accordingly yeah?

Reply:
Yeah I think that's right! I've never emitted an event in JS though so I'm not 100%.